cmake/sca/eclair/sca.cmake

```diff
@@ -0,0 +1,143 @@
+      set(ECLAIR_ANALYSIS_ECL_DIR ${ECLAIR_USER_RULESET_PATH})
+  else()
+    set(ECLAIR_ANALYSIS_ECL_DIR ${APPLICATION_CONFIG_DIR}/${ECLAIR_USER_RULESET_PATH})
+  endif()
+endif()
+# ECLAIR report
+if (ECLAIR_METRICS_TAB)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-metrics_tab=${ECLAIR_OUTPUT_DIR}/metrics")
+endif()
+if (ECLAIR_REPORTS_TAB)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-reports_tab=${ECLAIR_OUTPUT_DIR}/reports")
+endif()
+if (ECLAIR_REPORTS_SARIF)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-reports_sarif=${ECLAIR_OUTPUT_DIR}/reports.sarif")
+endif()
+if (ECLAIR_SUMMARY_TXT)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-summary_txt=${ECLAIR_OUTPUT_DIR}/summary_txt")
+endif()
+if (ECLAIR_SUMMARY_DOC)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-summary_doc=${ECLAIR_OUTPUT_DIR}/summary_doc")
+endif()
+if (ECLAIR_SUMMARY_ODT)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-summary_odt=${ECLAIR_OUTPUT_DIR}/summary_odt")
+endif()
+if (ECLAIR_FULL_TXT_ALL_AREAS)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-setq=report_areas,areas")
+endif()
+if (ECLAIR_FULL_TXT_FIRST_AREA)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-setq=report_areas,first_area")
+endif()
+if (ECLAIR_FULL_TXT)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-full_txt=${ECLAIR_OUTPUT_DIR}/txt")
+endif()
+if (ECLAIR_FULL_DOC_ALL_AREAS)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-setq=report_areas,areas")
+endif()
+if (ECLAIR_FULL_DOC_FIRST_AREA)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-setq=report_areas,first_area")
+endif()
+if (ECLAIR_FULL_DOC)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-full_doc=${ECLAIR_OUTPUT_DIR}/doc")
+endif()
+if (ECLAIR_FULL_ODT)
+  set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "${ECLAIR_REPORT_ADDITIONAL_OPTIONS}" "-full_odt=${ECLAIR_OUTPUT_DIR}/odt")
+endif()
+
+message(STATUS "ECLAIR outputs have been written to: ${ECLAIR_OUTPUT_DIR}")
+message(STATUS "ECLAIR ECB files have been written to: ${ECLAIR_DATA_DIR}")
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E remove_directory ${ECLAIR_OUTPUT_DIR}
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${ECLAIR_OUTPUT_DIR}
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${ECLAIR_DATA_DIR}
+  COMMAND_ERROR_IS_FATAL ANY
+  COMMAND_ECHO STDOUT
+)
+
+# configure the camke script which will be used to replace the compiler call with the eclair_env
+# call which calls the compiler and to generate analysis files.
+configure_file(${CMAKE_CURRENT_LIST_DIR}/eclair.template ${CMAKE_BINARY_DIR}/eclair.cmake @ONLY)
+
+set(launch_environment ${CMAKE_COMMAND} -P ${CMAKE_BINARY_DIR}/eclair.cmake --)
```

Comment:
I will move the file in the eclair out put folder :  sca/eclair, due to the fact that CMAKE_CURRENT_BINARY_DIR would also be the top-level because sca.cmake is processed on the top-level directory